lib/autoproj/python.rb

```diff
@@ -0,0 +1,317 @@
+        end
+
+        # Activate configuration for python in the autoproj configuration
+        # @return [String,String] python path and python version
+        def self.activate_python(ws: Autoproj.workspace,
+                                 bin: nil,
+                                 version: nil)
+            bin, version = resolve_python(ws: ws, bin: bin, version: version)
+            ws.config.set('python_executable', bin, true)
+            ws.config.set('python_version', version, true)
+
+            ws.osdep_suffixes << "python#{$1}" if version =~ /^([0-9]+)\./
+
+            rewrite_python_shims(bin, ws.root_dir)
+            rewrite_pip_shims(bin, ws.root_dir)
+            [bin, version]
+        end
+
+        def self.deactivate_python(ws: Autoproj.workspace)
+            remove_python_shims(ws.root_dir)
+            remove_pip_shims(ws.root_dir)
+            ws.config.reset('python_executable')
+            ws.config.reset('python_version')
+        end
+
+        # Allow to update the PYTHONPATH for package if autoproj configuration
+        # USE_PYTHON is set to true.
+        # Then tries to guess the python binary from Autobuild.programs['python']
+        # and system's default setting
+        # @param [Autobuild::Package] pkg
+        # @param [Autoproj.workspace] ws Autoproj workspace
+        # @param [String] bin Path to a custom python version
+        # @param [String] version version constraint for python executable
+        # @return tuple of [executable, version, site-packages path] if set,
+        #    otherwise nil
+        def self.activate_python_path(pkg,
+                                 ws: Autoproj.workspace,
+                                 bin: nil,
+                                 version: nil)
+            return unless ws.config.get('USE_PYTHON', nil)
+
+            bin, version = resolve_python(ws: ws, bin: bin, version: version)
+            path = File.join(pkg.prefix, "lib",
+                             "python#{version}", "site-packages")
+            pkg.env_add_path 'PYTHONPATH', path
+
+            [bin, version, path]
+        end
+
+        def self.prepare_venv(ws: Autoproj.workspace, bin: nil, version: nil,
+                              venv_name: ".autoproj_python_venv")
+
+          python_bin, = resolve_python(ws: ws, bin: bin, version: version)
+          cmd = "#{python_bin} -m pip install --user -U virtualenv"
+          msg, status = Open3.capture2e(cmd)
+          unless status.success?
+              raise "Autoproj::Python.prepare_venv installation of "\
+                  " virtualenv failed: '#{msg}'"
+          end
+
+          cmd = "#{python_bin} -m virtualenv #{venv_name}"
+          msg, status = Open3.capture2e(cmd)
+          unless status.success?
+              raise "Autoproj::Python.prepare_venv preparation of"\
+                  " virtual env '#{venv_name}' failed: '#{msg}'"
+          end
+
+          ws.env.set('VIRTUAL_ENV_DISABLE_PROMPT', 1)
```

Comment:
What is this for? I don't see it being used anywhere

Reply:
You do not see the special prompt from virtualenv, since it disables the prompt.

Reply:
Then shouldn't this always be set? The current implementation sets this only when `USE_PYTHON` does not exist

Reply:
Sorry, but I cannot follow you on this one? It is only set, when the virtual env is in use. At least that would be the intention of the current code, and this should only be the case when USE_PYTHON is set?

Reply:
The current code sets `VIRTUAL_ENV_DISABLE_PROMPT` [when `setup_python_configuration_options` is called](https://github.com/rock-core/autoproj/blob/40fcf063f8c62f3b8e29ae3b48de0cd5510d47e8/lib/autoproj/package_managers/pip_manager.rb#L41).  But that only gets called if `USE_PYTHON` does not exist (it won't be called if `USE_PYTHON` exists AND is `true`)